metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -303,14 +303,23 @@ object MetalsEnrichments
 
   implicit class XtensionAbsolutePathBuffers(path: AbsolutePath) {
 
-    def hasScalaFiles: Boolean = {
-      def isScalaDir(file: File): Boolean = {
+    def isScalaProject(extendSearchToScriptsAndSbt: Boolean = true): Boolean = {
+      val isScala: String => Boolean =
+        if (extendSearchToScriptsAndSbt) _.isScalaFilename else _.isScala
+      val directoriesToCheck = Set("test", "src", "it")
```

Comment:
Will work for the first one, since we take dirs that either are `directoriesToCheck` or have `directoriesToCheck` inside, won't work with the second example. But it doesn't matter that much, since, as you noticed, the moment user opens a `.scala` (or `.sc`, `.sbt`) file corresponding workspace folder will get transformed into a Scala project. That's similar to how the extension is triggered so it seems reasonable. We can adjust the heuristic, I just think it'd be nice to avoid a check that could run for too long.

Reply:
So 👍 means we want to try a more broad search or not?

Reply:
I think the +1 was that we have a special fallback that should work. I don't think we need a broad search